lib/down/net_http.rb

```diff
@@ -155,6 +157,10 @@ def open_uri(uri, options, follows_remaining:)
         raise ResponseError.new("Invalid Redirect URI: #{exception.uri}", response: response)
       end
 
+      # do not leak credentials on redirect
+      options.delete("Authorization") unless auth_on_redirect
+      options.delete(:http_basic_authentication) unless auth_on_redirect
```

Comment:
same question in a different place - should there be some logic to keep the creds if it's a relative redirect?

Reply:
there's not a good way to preserve on a relative redirect - i can check to see if the host is the same, but that doesn't help from a test perspective. for now, i'm keping the logic as-is.